Sources/CardPayments/CardClient.swift

```diff
@@ -74,13 +74,20 @@ public class CardClient: NSObject {
             do {
                 let result = try await checkoutOrdersAPI.confirmPaymentSource(cardRequest: request)
                 
-                if let url: String = result.links?.first(where: { $0.rel == "payer-action" })?.href {
+                if result.status == "PAYER_ACTION_REQUIRED",
+                let url = result.links?.first(where: { $0.rel == "payer-action" })?.href {
+                    
+                    guard let url = URL(string: url), url.absoluteString.contains("flow=3ds"), url.absoluteString.contains("helios") else {
+                        self.notifyFailure(with: CardClientError.threeDSecureURLError)
```

Comment:
I'm wondering now if we should just check for 
`if let url = result.links?.first(where: { $0.rel == "payer-action" })?.href, url.contains("flow=3ds"), url.contains("helios") {` as conditions for going into 3DS flow and have the same guard in the 3DS for URL.

Reply:
I think if we do want to reuse the startThreeDSecureChallenge function to return to different notify delegate functions for vaulting without purchase, we should not include result.status, I think for vault without purchase, it is "APPROVED". I have to double check my notes.

Reply:
I got it. What we currently makes sense, we are just validating the status field with 3DS team.